dynamic_rest/filters.py

```diff
@@ -517,7 +518,15 @@ def _filter_queryset(
         )
 
         if query:
-            queryset = queryset.filter(query)
+            # Convert internal django ValidationError to APIException-based one
+            # in order to resolve validation errors from 500 status code to
+            # 400.
+            try:
+                queryset = queryset.filter(query)
+            except InternalValidationError as e:
+                raise ValidationError(
+                    dict(e) if hasattr(self, 'error_dict') else list(e)
```

Comment:
Yes, see https://github.com/django/django/blob/master/django/core/exceptions.py#L167-L170 for details. This is report `__str__` method because we need actual messages.